### ISSUE TITLE
Apply flash attention

### ISSUE THREAD

**Issue by qwopqwop200** [fpgaminer/GPTQ-triton#9]
https://github.com/qwopqwop200/GPTQ-for-LLaMa/commit/47dd6b32b7c5cdf07ac14b395fe04c92303efc25
I experienced an approximate 5% speedup by simply applying flash attention as follows

**Comment by fpgaminer** [comment]
I believe you need to pass attention mask there for it to be correct. And if you do, then at least on PyTorch 2.0 it fails because it doesn't have a non-math implementation with attention mask support :(

My plan was to either use xformers (which is a dependency nightmare), port over a Triton implementation of Flash, or just wait for PyTorch to improve support.
